src/modules/app/containers/Switcher.js

```diff
@@ -79,38 +97,75 @@ export class Switcher extends React.Component<Props> {
         return Switcher.renderFailureLoadingComponents(categoriesPayload) || <SearchPage />
       case CATEGORIES_REDIRECT_ROUTE:
         return Switcher.renderFailureLoadingComponents(categoriesPayload) || <LoadingSpinner />
-      default:
-        return <Failure error={'Route not found'} />
+      case NOT_FOUND:
+        // The only possibility to be in the NOT_FOUND route is if we have "/:param" as path and the param is neither
+        // "disclaimer" nor a city, so we want to show an error that the param is not an available city
+        if (param) {
+          const error = new CityNotFoundError({city: param})
+          return <FailureSwitcher error={error} />
+        }
     }
+    throw new Error(`For the route ${currentRoute} was no content selected in the Switcher which should be displayed.`)
   }
 
   render () {
-    const {viewportSmall, currentRoute} = this.props
+    const {city, citiesPayload, language, languages, currentRoute, viewportSmall} = this.props
+
+    // The current city is invalid, so we want to show an error
+    if (city && Array.isArray(citiesPayload.data) &&
+      !citiesPayload.data.find(_city => _city instanceof CityModel && _city.code === city)) {
```

Comment:
nope, but it is requested by flow because we could store anything there....but I think in the mid term we could do it like this: https://flow.org/en/docs/react/redux/